graphql_compiler/tests/test_data_tools/data_tool.py

```diff
@@ -15,58 +17,211 @@
+        ),
+        'Animal_ParentOf': (
+            {
+                'from_uuid': uuids['A1'],
+                'to_uuid': uuids['A1'],
+            }, {
+                'from_uuid': uuids['A1'],
+                'to_uuid': uuids['A2'],
+            }, {
+                'from_uuid': uuids['A1'],
+                'to_uuid': uuids['A3'],
+            }, {
+                'from_uuid': uuids['A3'],
+                'to_uuid': uuids['A4'],
+            }
+        )
+    }
+
+    # Find the class name of each vertex uuid
+    uuid_to_class_name = {}
+    for vertex_name, values in six.iteritems(vertex_values):
+        for value in values:
+            if value['uuid'] in uuid_to_class_name:
+                raise AssertionError(u'Duplicate uuid found {}'.format(value['uuid']))
+            uuid_to_class_name[value['uuid']] = vertex_name
+
+    return vertex_values, edge_values, uuid_to_class_name
+
+
 def generate_orient_snapshot_data(client):
     """Create OrientDB test DB from the SQL commands file for snapshot testing."""
     project_root = path.dirname(path.abspath(__file__))
     sql_files = glob(path.join(project_root, 'snapshot_data/*.sql'))
-    _load_sql_files_to_orient_client(client, sql_files)
 
-
-def generate_orient_integration_data(client):
-    """Create OrientDB test DB from the SQL commands file for snapshot testing."""
-    project_root = path.dirname(path.abspath(__file__))
-    sql_files = glob(path.join(project_root, 'integration_data/*.sql'))
     _load_sql_files_to_orient_client(client, sql_files)
 
 
-def generate_neo4j_integration_data(client):
-    """Create Neo4j test DB from the SQL commands file for integration testing."""
-    project_root = path.dirname(path.abspath(__file__))
-    neo4j_files = glob(path.join(project_root, 'integration_data/create_neo4j_integration.cypher'))
-    _load_neo4j_files_to_neo4j_client(client, neo4j_files)
+def _validate_name(name):
+    """Check conservatively for allowed vertex and field names."""
+    if not isinstance(name, six.string_types):
+        raise AssertionError(u'Expected string name. Received {}'.format(name))
+    allowed_characters = string.ascii_uppercase + string.ascii_lowercase + string.digits + '_'
+    if any(c not in allowed_characters for c in str(name)):
+        raise AssertionError(u'Name {} contains disallowed characters'.format(name))
+
+
+def _write_orient_equality(field_name, field_value):
+    """Write a '{} = {}' statement to be used in queries, validating against SQL injection."""
+    _validate_name(field_name)
+    if isinstance(field_value, six.string_types):
+        allowed_characters = string.ascii_uppercase + string.ascii_lowercase + string.digits + '-_ '
+        if any(c not in allowed_characters for c in str(field_value)):
+            raise AssertionError(u'String value {} contains disallowed characters'
+                                 .format(field_value))
+    elif isinstance(field_value, (six.integer_types, float, datetime.date)):
```

Comment:
I think `six.integer_types` is already a tuple, and nested tuples don't work correctly in `isinstance` -- they aren't flattened.

```
$ python
Python 2.7.16 (default, Sep  2 2019, 11:59:44) 
[GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.46.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
>>> six.integer_types
(<type 'int'>, <type 'long'>)
>>> exit()

$ python3
Python 3.7.4 (default, Sep  7 2019, 18:27:02) 
[Clang 10.0.1 (clang-1001.0.46.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import six
>>> six.integer_types
(<class 'int'>,)
```

Use `six.integer_types + (float, datetime.date)` instead.

Reply:
https://docs.python.org/3/library/functions.html#isinstance `isinstance` takes nested tuples.

Reply:
I thought it didn't work in python 2, but I stand corrected -- the py2 version of that page says the same thing. My bad.